### ISSUE TITLE
[Bug] Unexpected token i in JSON at position 1

### ISSUE THREAD

**Issue by kiwimato** [GitTools/GitVersion#2567]
**Describe the bug**
Azure DevOps task fails with error `Unexpected token i in JSON at position 1`  even though GitVersion seems to complete successfully. Tried running it locally without an issue and it works flawlessly for other repositories. It also used to work for this repository as well.

## Expected Behavior
Should work.

## Actual Behavior
Task fails after saying "Done writing" with error `Unexpected token i in JSON at position 1`. Checked the JSON just before the message and it's valid JSON.

## Steps to Reproduce
To be honest I don't know how to reproduce it, since this happens only on a specific repository. Works on other repositories.
I also tried running GitVersion locally on the same repository andit works:
```
  INFO [01/27/21 14:46:56:87] Done writing

C:\Users\X\Documents\code\projects>echo %errorlevel%
0

```

## Context
All of a sudden I can't run that specific pipeline anymore since it used to work before.

## Your Environment
Azure Devops, self host agents or Azure, same issue. Also for other repositories GitVersion works well on the same agents.

- Version Used:
- Operating System and version (Windows 10, Ubuntu 18.04):
- Link to your project:
- Link to your CI build (if appropriate):

## Debug Logs
```
2021-01-28T11:13:33.4728806Z Executing GenerateSetVersionMessage for 'AzurePipelines'.
2021-01-28T11:13:33.4737446Z ##[debug]Update build number for build: 68047 to: 1.12.1-PullRequest2464.6 at backend.
2021-01-28T11:13:33.5194556Z ##[debug]Processed: ##vso[build.updatebuildnumber]1.12.1-PullRequest2464.6
2021-01-28T11:13:33.5195137Z Executing GenerateBuildLogOutput for 'AzurePipelines'.
2021-01-28T11:13:33.5196127Z ##[debug]Processed: ##vso[task.setvariable variable=GitVersion.Major]1
2021-01-28T11:13:33.5203384Z ##[debug]Processed: ##vso[task.setvariable variable=GitVersion.Major;isOutput=true]1
2021-01-28T11:13:33.5204031Z ##[debug]Processed: ##vso[task.setvariable variable=GitVersion.Minor]12
2021-01-28T11:13:33.5204954Z ##[debug]Processed: ##vso[task.setvariable variable=GitVersion.Minor;isOutput=true]12
2021-01-28T11:13:33.5205499Z ##[debug]Processed: ##vso[task.setvariable variable=GitVersion.Patch]1
2021-01-28T11:13:33.5206302Z ##[debug]Processed: ##vso[task.setvariable variable=GitVersion.Patch;isOutput=true]1
2021-01-28T11:13:33.5206913Z ##[debug]Processed: ##vso[task.setvariable variable=GitVersion.PreReleaseTag]PullRequest2464.6
2021-01-28T11:13:33.5207815Z ##[debug]Processed: ##vso[task.setvariable variable=GitVersion.PreReleaseTag;isOutput=true]PullRequest2464.6
2021-01-28T11:13:33.5208449Z ##[debug]Processed: ##vso[task.setvariable variable=GitVersion.PreReleaseTagWithDash]-PullRequest2464.6
2021-01-28T11:13:33.5209448Z ##[debug]Processed: ##vso[task.setvariable variable=GitVersion.PreReleaseTagWithDash;isOutput=true]-PullRequest2464.6
2021-01-28T11:13:33.5210091Z ##[debug]Processed: ##vso[task.setvariable variable=GitVersion.PreReleaseLabel]PullRequest2464
2021-01-28T11:13:33.5210988Z ##[debug]Processed: ##vso[task.setvariable variable=GitVersion.PreReleaseLabel;isOutput=true]PullRequest2464
2021-01-28T11:13:33.5211626Z ##[debug]Processed: ##vso[task.setvariable variable=GitVersion.PreReleaseLabelWithDash]-PullRequest2464
2021-01-28T11:13:33.5212554Z ##[debug]Processed: ##vso[task.setvariable variable=GitVersion.PreReleaseLabelWithDash;isOutput=true]-PullRequest2464
2021-01-28T11:13:33.5213407Z ##[debug]Processed: ##vso[task.setvariable variable=GitVersion.PreReleaseNumber]6
2021-01-28T11:13:33.5214299Z ##[debug]Processed: ##vso[task.setvariable variable=GitVersion.PreReleaseNumber;isOutput=true]6
2021-01-28T11:13:33.5214893Z ##[debug]Processed: ##vso[task.setvariable variable=GitVersion.WeightedPreReleaseNumber]30006
2021-01-28T11:13:33.5215799Z ##[debug]Processed: ##vso[task.setvariable variable=GitVersion.WeightedPreReleaseNumber;isOutput=true]30006
2021-01-28T11:13:33.5216365Z ##[debug]Processed: ##vso[task.setvariable variable=GitVersion.BuildMetaData]
2021-01-28T11:13:33.5217646Z ##[debug]Processed: ##vso[task.setvariable variable=GitVersion.BuildMetaData;isOutput=true]
2021-01-28T11:13:33.5218217Z ##[debug]Processed: ##vso[task.setvariable variable=GitVersion.BuildMetaDataPadded]
2021-01-28T11:13:33.5219083Z ##[debug]Processed: ##vso[task.setvariable variable=GitVersion.BuildMetaDataPadded;isOutput=true]
2021-01-28T11:13:33.5219815Z ##[debug]Processed: ##vso[task.setvariable variable=GitVersion.FullBuildMetaData]Branch.pull-2464-merge.Sha.b94d05ec0f36072cc01a8190bc6a350094b2e6c9
2021-01-28T11:13:33.5220944Z ##[debug]Processed: ##vso[task.setvariable variable=GitVersion.FullBuildMetaData;isOutput=true]Branch.pull-2464-merge.Sha.b94d05ec0f36072cc01a8190bc6a350094b2e6c9
2021-01-28T11:13:33.5221587Z ##[debug]Processed: ##vso[task.setvariable variable=GitVersion.MajorMinorPatch]1.12.1
2021-01-28T11:13:33.5222452Z ##[debug]Processed: ##vso[task.setvariable variable=GitVersion.MajorMinorPatch;isOutput=true]1.12.1
2021-01-28T11:13:33.5223055Z ##[debug]Processed: ##vso[task.setvariable variable=GitVersion.SemVer]1.12.1-PullRequest2464.6
2021-01-28T11:13:33.5224144Z ##[debug]Processed: ##vso[task.setvariable variable=GitVersion.SemVer;isOutput=true]1.12.1-PullRequest2464.6
2021-01-28T11:13:33.5224764Z ##[debug]Processed: ##vso[task.setvariable variable=GitVersion.LegacySemVer]1.12.1-PullRequest2464-6
2021-01-28T11:13:33.5225697Z ##[debug]Processed: ##vso[task.setvariable variable=GitVersion.LegacySemVer;isOutput=true]1.12.1-PullRequest2464-6
2021-01-28T11:13:33.5226362Z ##[debug]Processed: ##vso[task.setvariable variable=GitVersion.LegacySemVerPadded]1.12.1-PullRequest2464-0006
2021-01-28T11:13:33.5227405Z ##[debug]Processed: ##vso[task.setvariable variable=GitVersion.LegacySemVerPadded;isOutput=true]1.12.1-PullRequest2464-0006
2021-01-28T11:13:33.5228012Z ##[debug]Processed: ##vso[task.setvariable variable=GitVersion.AssemblySemVer]1.12.1.0
2021-01-28T11:13:33.5228887Z ##[debug]Processed: ##vso[task.setvariable variable=GitVersion.AssemblySemVer;isOutput=true]1.12.1.0
2021-01-28T11:13:33.5229489Z ##[debug]Processed: ##vso[task.setvariable variable=GitVersion.AssemblySemFileVer]1.12.1.0
2021-01-28T11:13:33.5230386Z ##[debug]Processed: ##vso[task.setvariable variable=GitVersion.AssemblySemFileVer;isOutput=true]1.12.1.0
2021-01-28T11:13:33.5230993Z ##[debug]Processed: ##vso[task.setvariable variable=GitVersion.FullSemVer]1.12.1-PullRequest2464.6
2021-01-28T11:13:33.5231931Z ##[debug]Processed: ##vso[task.setvariable variable=GitVersion.FullSemVer;isOutput=true]1.12.1-PullRequest2464.6
2021-01-28T11:13:33.5232752Z ##[debug]Processed: ##vso[task.setvariable variable=GitVersion.InformationalVersion]1.12.1-PullRequest2464.6+Branch.pull-2464-merge.Sha.b94d05ec0f36072cc01a8190bc6a350094b2e6c9
2021-01-28T11:13:33.5233974Z ##[debug]Processed: ##vso[task.setvariable variable=GitVersion.InformationalVersion;isOutput=true]1.12.1-PullRequest2464.6+Branch.pull-2464-merge.Sha.b94d05ec0f36072cc01a8190bc6a350094b2e6c9
2021-01-28T11:13:33.5234680Z ##[debug]Processed: ##vso[task.setvariable variable=GitVersion.BranchName]pull/2464/merge
2021-01-28T11:13:33.5235577Z ##[debug]Processed: ##vso[task.setvariable variable=GitVersion.BranchName;isOutput=true]pull/2464/merge
2021-01-28T11:13:33.5236188Z ##[debug]Processed: ##vso[task.setvariable variable=GitVersion.EscapedBranchName]pull-2464-merge
2021-01-28T11:13:33.5237232Z ##[debug]Processed: ##vso[task.setvariable variable=GitVersion.EscapedBranchName;isOutput=true]pull-2464-merge
2021-01-28T11:13:33.5237866Z ##[debug]Processed: ##vso[task.setvariable variable=GitVersion.Sha]b94d05ec0f36072cc01a8190bc6a350094b2e6c9
2021-01-28T11:13:33.5238821Z ##[debug]Processed: ##vso[task.setvariable variable=GitVersion.Sha;isOutput=true]b94d05ec0f36072cc01a8190bc6a350094b2e6c9
2021-01-28T11:13:33.5239477Z ##[debug]Processed: ##vso[task.setvariable variable=GitVersion.ShortSha]b94d05e
2021-01-28T11:13:33.5240343Z ##[debug]Processed: ##vso[task.setvariable variable=GitVersion.ShortSha;isOutput=true]b94d05e
2021-01-28T11:13:33.5240954Z ##[debug]Processed: ##vso[task.setvariable variable=GitVersion.NuGetVersionV2]1.12.1-pullrequest2464-0006
2021-01-28T11:13:33.5241931Z ##[debug]Processed: ##vso[task.setvariable variable=GitVersion.NuGetVersionV2;isOutput=true]1.12.1-pullrequest2464-0006
2021-01-28T11:13:33.5242574Z ##[debug]Processed: ##vso[task.setvariable variable=GitVersion.NuGetVersion]1.12.1-pullrequest2464-0006
2021-01-28T11:13:33.5243525Z ##[debug]Processed: ##vso[task.setvariable variable=GitVersion.NuGetVersion;isOutput=true]1.12.1-pullrequest2464-0006
2021-01-28T11:13:33.5244172Z ##[debug]Processed: ##vso[task.setvariable variable=GitVersion.NuGetPreReleaseTagV2]pullrequest2464-0006
2021-01-28T11:13:33.5245122Z ##[debug]Processed: ##vso[task.setvariable variable=GitVersion.NuGetPreReleaseTagV2;isOutput=true]pullrequest2464-0006
2021-01-28T11:13:33.5245756Z ##[debug]Processed: ##vso[task.setvariable variable=GitVersion.NuGetPreReleaseTag]pullrequest2464-0006
2021-01-28T11:13:33.5246711Z ##[debug]Processed: ##vso[task.setvariable variable=GitVersion.NuGetPreReleaseTag;isOutput=true]pullrequest2464-0006
2021-01-28T11:13:33.5248072Z ##[debug]Processed: ##vso[task.setvariable variable=GitVersion.VersionSourceSha]a5d68cf86a6acbfcd294217a4048895c4b55c6f4
2021-01-28T11:13:33.5249112Z ##[debug]Processed: ##vso[task.setvariable variable=GitVersion.VersionSourceSha;isOutput=true]a5d68cf86a6acbfcd294217a4048895c4b55c6f4
2021-01-28T11:13:33.5249759Z ##[debug]Processed: ##vso[task.setvariable variable=GitVersion.CommitsSinceVersionSource]6
2021-01-28T11:13:33.5250674Z ##[debug]Processed: ##vso[task.setvariable variable=GitVersion.CommitsSinceVersionSource;isOutput=true]6
2021-01-28T11:13:33.5251299Z ##[debug]Processed: ##vso[task.setvariable variable=GitVersion.CommitsSinceVersionSourcePadded]0006
2021-01-28T11:13:33.5252240Z ##[debug]Processed: ##vso[task.setvariable variable=GitVersion.CommitsSinceVersionSourcePadded;isOutput=true]0006
2021-01-28T11:13:33.5252816Z ##[debug]Processed: ##vso[task.setvariable variable=GitVersion.UncommittedChanges]0
2021-01-28T11:13:33.5253713Z ##[debug]Processed: ##vso[task.setvariable variable=GitVersion.UncommittedChanges;isOutput=true]0
2021-01-28T11:13:33.5254275Z ##[debug]Processed: ##vso[task.setvariable variable=GitVersion.CommitDate]2021-01-28
2021-01-28T11:13:33.5255244Z ##[debug]Processed: ##vso[task.setvariable variable=GitVersion.CommitDate;isOutput=true]2021-01-28
2021-01-28T11:13:33.5255520Z {
2021-01-28T11:13:33.5255672Z   "Major": 1,
2021-01-28T11:13:33.5255828Z   "Minor": 12,
2021-01-28T11:13:33.5256002Z   "Patch": 1,
2021-01-28T11:13:33.5256194Z   "PreReleaseTag": "PullRequest2464.6",
2021-01-28T11:13:33.5256457Z   "PreReleaseTagWithDash": "-PullRequest2464.6",
2021-01-28T11:13:33.5256716Z   "PreReleaseLabel": "PullRequest2464",
2021-01-28T11:13:33.5256962Z   "PreReleaseLabelWithDash": "-PullRequest2464",
2021-01-28T11:13:33.5257203Z   "PreReleaseNumber": 6,
2021-01-28T11:13:33.5257412Z   "WeightedPreReleaseNumber": 30006,
2021-01-28T11:13:33.5257640Z   "BuildMetaData": "",
2021-01-28T11:13:33.5257845Z   "BuildMetaDataPadded": "",
2021-01-28T11:13:33.5258144Z   "FullBuildMetaData": "Branch.pull-2464-merge.Sha.b94d05ec0f36072cc01a8190bc6a350094b2e6c9",
2021-01-28T11:13:33.5258448Z   "MajorMinorPatch": "1.12.1",
2021-01-28T11:13:33.5258680Z   "SemVer": "1.12.1-PullRequest2464.6",
2021-01-28T11:13:33.5259278Z   "LegacySemVer": "1.12.1-PullRequest2464-6",
2021-01-28T11:13:33.5259554Z   "LegacySemVerPadded": "1.12.1-PullRequest2464-0006",
2021-01-28T11:13:33.5259795Z   "AssemblySemVer": "1.12.1.0",
2021-01-28T11:13:33.5260027Z   "AssemblySemFileVer": "1.12.1.0",
2021-01-28T11:13:33.5260277Z   "FullSemVer": "1.12.1-PullRequest2464.6",
2021-01-28T11:13:33.5260634Z   "InformationalVersion": "1.12.1-PullRequest2464.6+Branch.pull-2464-merge.Sha.b94d05ec0f36072cc01a8190bc6a350094b2e6c9",
2021-01-28T11:13:33.5260974Z   "BranchName": "pull/2464/merge",
2021-01-28T11:13:33.5261223Z   "EscapedBranchName": "pull-2464-merge",
2021-01-28T11:13:33.5261473Z   "Sha": "b94d05ec0f36072cc01a8190bc6a350094b2e6c9",
2021-01-28T11:13:33.5261759Z   "ShortSha": "b94d05e",
2021-01-28T11:13:33.5261987Z   "NuGetVersionV2": "1.12.1-pullrequest2464-0006",
2021-01-28T11:13:33.5262263Z   "NuGetVersion": "1.12.1-pullrequest2464-0006",
2021-01-28T11:13:33.5262540Z   "NuGetPreReleaseTagV2": "pullrequest2464-0006",
2021-01-28T11:13:33.5262808Z   "NuGetPreReleaseTag": "pullrequest2464-0006",
2021-01-28T11:13:33.5263102Z   "VersionSourceSha": "a5d68cf86a6acbfcd294217a4048895c4b55c6f4",
2021-01-28T11:13:33.5263379Z   "CommitsSinceVersionSource": 6,
2021-01-28T11:13:33.5263620Z   "CommitsSinceVersionSourcePadded": "0006",
2021-01-28T11:13:33.5263865Z   "UncommittedChanges": 0,
2021-01-28T11:13:33.5264066Z   "CommitDate": "2021-01-28"
2021-01-28T11:13:33.5264240Z }
2021-01-28T11:13:33.5264432Z   INFO [01/28/21 12:13:33:45] Done writing 
2021-01-28T11:13:33.5264771Z ##[debug]task result: Failed
2021-01-28T11:13:33.5265401Z ##[error]SyntaxError: Unexpected token i in JSON at position 1
2021-01-28T11:13:33.5266445Z ##[debug]Processed: ##vso[task.issue type=error;]SyntaxError: Unexpected token i in JSON at position 1
2021-01-28T11:13:33.5269203Z ##[debug]Processed: ##vso[task.complete result=Failed;done=true;]SyntaxError: Unexpected token i in JSON at position 1
2021-01-28T11:13:33.5296255Z ##[section]Async Command Start: Update Build Number
2021-01-28T11:13:33.7540949Z Update build number to 1.12.1-PullRequest2464.6 for build 68047
2021-01-28T11:13:33.7541306Z ##[section]Async Command End: Update Build Number
2021-01-28T11:13:33.7542864Z ##[section]Finishing: Use GitVersion
```

**Comment by asbjornu** [comment]
How does your `.gitattributes` look in the faulty repository? Could it be a problem with BOM, EOL or similar?

**Comment by mihai-cindea** [comment]
Just checked and there's no `.gitattributes` file in the repository.

**Comment by mihai-cindea** [comment]
There's no BOM or EOL issues either as far as I can see. Also the weird fact is that is that it first ran successfully on the same commit, so it seems it started failing without any changes to the repository, at least the main branch (same commit hash)
![image](https://user-images.githubusercontent.com/56123381/106257544-5be55400-6225-11eb-8685-793f0ab71e58.png)

**Comment by mihai-cindea** [comment]
In case anyone hits this issue in the future: I managed to avoid it by rewriting git history: Created a new repository from scratch and simply added everything in 1 commit.

**Comment by mihai-cindea** [comment]
It broke again, but finally found what broke it: 
A branch name containing `{}` characters like: `TICKET-contacts-{id}-email` throws the error: 
```
##[error]SyntaxError: Unexpected token i in JSON at position 1
```
This error now makes sense, because `i` is at position 1 in `{id}`.

After I removed the branch upstream and ran again it worked successfully. I think this might be an issue on how GitVersion handles the branch name internally.

**Comment by asbjornu** [comment]
This sounds like something that should be pretty easy to provoke in a test. Are you up for providing us with a PR with such a test, and possibly also a fix? 🙏🏼

**Comment by mihai-cindea** [comment]
Sure, if you could please pinpoint me to some basics since I'm not familiar with your code base.

**Comment by arturcic** [comment]
@mihai-cindea you can have a look here https://github.com/GitTools/GitVersion/tree/main/src/GitVersion.Core.Tests/IntegrationTests for examples

**Comment by arturcic** [comment]
You can put the test in https://github.com/GitTools/GitVersion/blob/main/src/GitVersion.Core.Tests/IntegrationTests/OtherScenarios.cs

**Comment by TomaszWegrzyn** [comment]
I encountered the same issue(It only occurred for single repository out of 40 😕). Something is wrong with versions 5.6.5 and higher. 

Setting `versionSpec: '5.6.4'` for my `gitversion/setup@0` task on Azure Pipelines is my solution for now.

**Comment by mihai-cindea** [comment]
@arturcic tried setting up a test case, but I see the `{}` characters are replaced with `-`:
```
A total of 1 test files matched the specified pattern.
  Failed NoSpecialCharactersBranch [138 ms]
  Error Message:
   Shouldly.ShouldAssertException : variables.FullSemVer
    should be
"1.0.1-branch-{param}.1+1"
    but was
"1.0.1-branch--param-.1+1"
    difference
Difference     |                                                         |                             |                       
               |                                                        \|/                           \|/                      
Index          | ...  3    4    5    6    7    8    9    10   11   12   13   14   15   16   17   18   19   20   21   22   23   
Expected Value | ...  .    1    -    b    r    a    n    c    h    -    {    p    a    r    a    m    }    .    1    +    1    
Actual Value   | ...  .    1    -    b    r    a    n    c    h    -    -    p    a    r    a    m    -    .    1    +    1    
Expected Code  | ...  46   49   45   98   114  97   110  99   104  45   123  112  97   114  97   109  125  46   49   43   49   
Actual Code    | ...  46   49   45   98   114  97   110  99   104  45   45   112  97   114  97   109  45   46   49   43   49   
  Stack Trace:
     at GitVersion.Core.Tests.GitToolsTestingExtensions.AssertFullSemver(RepositoryFixtureBase fixture, String fullSemver, Config configuration, IRepository repository, String commitId, Boolean onlyTrackedBranches, String targetBranch) in /home/mihai/code/GitVersion/src/GitVersion.Core.Tests/Extensions/GitToolsTestingExtensions.cs:line 132
   at GitVersion.Core.Tests.IntegrationTests.OtherScenarios.NoSpecialCharactersBranch() in /home/mihai/code/GitVersion/src/GitVersion.Core.Tests/IntegrationTests/OtherScenarios.cs:line 146
  Standard Output Messages:
 Initialized empty Git repository in /tmp/TestRepositories/f70afaa9-5b75-4d3f-b698-cea4b6648bb9/.git/
```

After I changed the assert, the tests passed. I think it's more of an issue on how the result is passed to Azure DevOps and not GitVersion since locally I can run it just fine, it doesn't error out.

**Comment by ASarco** [comment]
I had the same problem, when running gitversion as a Github action, the issue was with one commit message:
`Changed version into ${revision} incremented base versions Patch, now 0.1.76`

So for some reason gitversion (but only when running in an action, or on Azure devops), thinks text with {} in comments and names is JSON, and obviously fails to parse it.

**Comment by dsbut** [comment]
I have also run into this problem provoked by a commit message that includes text that GitVersion is apparently misinterpreting as JSON. The comment is similar to the following:

"Implement a `GET /api/{version}/SomeResourceType/{resourceId}` endpoint

The commit message has propagated to my master branch, and I won't be able to fix this without a complicated history re-write.

**Comment by oliverroer** [comment]
I've encounted the same problem in Azure DevOps using the [GitTools extension](https://marketplace.visualstudio.com/items?itemName=gittools.gittools).

A commit contained the `{` character, which tripped up GitVersion.

As a temporary workaround, I tried to follow the solution posted by @TomaszWegrzyn - however, version `5.6.3`, not `5.6.4` seemed to do the job in my case. 🤔

**Comment by michelefa1988** [comment]
I had the same issue. Was using Azure devops GitTools extension. I had set my Gitversion to `5.x.x` but had to fix to version `5.6.3` 

I also had no branches or comments which included `{` in the name

**Comment by JackSinclairT** [comment]
Same issue here with us using this in Azure Devops for our yaml pipelines. We've had to downgrade our version 5.6.3 to get this working

**Comment by Shazwazza** [comment]
This issue occurs for me if I add the `showConfig` flag like this:

```yml
- task: gitversion/execute@0
  displayName: Determine Version
  inputs:
    additionalArguments: '/showConfig'
```

without that argument it works.

**Comment by wpschaub** [comment]
Our Azure DevOps YAML pipelines have also started failing with a similar error:
 ...
INFO [06/07/21 23:34:50:18] Done writing 
##[error]SyntaxError: Unexpected token 
 in JSON at position 17

Forcing the version to 5.6.3 unblocked the pipelines. 

Any suggested steps to troubleshoot the issue? We would like to return to the latest version.

**Comment by asbjornu** [comment]
It seems like this issue only occurs in the AzDO Task, in which case GitTools/actions#431 and GitTools/actions#370 may be related (and also, still unfixed). Since I don't use AzDO for anything myself, I have little incentive or opportunity to debug and fix this issue. If someone else can venture into the AzDO Task codebase, figure out and fix the problem, pull requests are warmly welcome. 🙏🏼

**Comment by wpschaub** [comment]
@asbjornu would it be feasible to beef up the error handling to optionally allow us to retrieve more detail around the syntax error. For example, if troubleshooting = true, dump the json file "as is", so that we can get a better idea of what is at position X and why the logic is failing?

I will see if I can re-produce and investigate the issue on our side.

**Comment by asbjornu** [comment]
@wpschaub, yes absolutely. You can just base it on the existing `/verbosity` argument, no new argument needed.

**Comment by bcowdery** [comment]
This is affecting our GitHubActions build using GitVersion 5.6.11

Offending git commit title has a `}` character in it due to a developer typo when merging a PR.

Here's an example commit that reproduces the issue:
```
commit e29df265ddbc84a30a2d99b0c7313aaefaf986e7
Author: bcowdery <bcowdery@pointyspoon.com>
Date:   Mon Sep 13 15:27:12 2021 -0600

    feat: [BA-215} Sample commit that breaks GitVersion (#502)

    Establishes that special characters in the commit title can bork JSON parsing
```

I was unable to reproduce this locally with command-line tools. This issue appears to only occur when running with GitHub Actions. We've rolled back to version 5.6.3 to resolve the issue.

For the sake of completeness, my GitVersion.yml looks like:

```
mode: Mainline
increment: Patch
tag-prefix: 'campaign/'
branches:
  main:
    regex: ^main$
    prevent-increment-of-merged-branch-version: true
    track-merge-target: false  
    tracks-release-branches: false
    is-release-branch: false
    is-mainline: true
    is-source-branch-for: [ 'feature', 'test', 'production' ]
  feature:
    regex: ^features?[/-]
    mode: ContinuousDelivery # use last tag, makes numbers predictable until merge
    tag: useBranchName
    increment: Inherit
  test:
    regex: ^test[/-]
    tag: ''
    increment: Inherit
    prevent-increment-of-merged-branch-version: true
    track-merge-target: false
    tracks-release-branches: false
    is-release-branch: false
    is-mainline: true
  production:
    regex: ^production[/-]
    tag: ''
    increment: Inherit
    prevent-increment-of-merged-branch-version: true
    track-merge-target: false
    tracks-release-branches: false
    is-release-branch: false
    is-mainline: true
ignore:
  sha: []
```

**Comment by asbjornu** [comment]
If rolling back GitVersion helps, I wonder how this can only occur in GitHub Actions. 🤔 Anyway, thanks for the information @bcowdery!

**Comment by bcowdery** [comment]
Rolling back to `5.6.3` introduced a regression in `mainline` versioning mode. There's an extra pre-release token being appended to my build number which is breaking NuGet versioning (  "NuGetVersion": "0.1.644-0658" ). 

Unfortunately it looks like rolling back isn't an option for us since the bug is affecting versions `5.6.4` and greater - it looks like I'm going to have to resort to a forced re-write of my git history to fix the offending commit.

Please let me know if there is anything I can do to help reproduce or fix this issue. Thanks!

**Comment by asbjornu** [comment]
@bcowdery, if you're able to reproduce this in an integration test, that would be great. I'm not sure whether anyone have attempted to make it yet. See this test as an example:

https://github.com/GitTools/GitVersion/blob/437f4423b2ff2876da65d3ad9016b0e9bf54a0cd/src/GitVersion.Core.Tests/IntegrationTests/VersionBumpingScenarios.cs#L36-L49

Otherwise, a public repository that is able to reproduce the error would be very useful.

**Comment by wpschaub** [comment]
Similarly, we have identified that the repos which are causing GitVersion to fail on our Azur DevOps side, contains "}" in commit messages.
![image](https://user-images.githubusercontent.com/12176555/134917885-d3705caa-2cb2-403c-8bce-f3e084fffdd9.png)

**Comment by einarth** [comment]
In case it helps, I can reproduce this issue using GitHub Actions in any repo (GitVersion 5.8.1) by just pushing a commit with a json-y message:
`git commit -m "{'jsonstring1': X}"`

output from action:
```
Run gittools/actions/gitversion/execute@v0.9.9
  with:
    useConfigFile: true
    configFilePath: GitVersion.yml
    updateAssemblyInfo: false
  env:
    DOTNET_CLI_TELEMETRY_OPTOUT: true
    DOTNET_NOLOGO: true
    DOTNET_ROOT: ../share/dotnet
Command: dotnet-gitversion /home/runner/work/drift-seasons/drift-seasons /output json /output buildserver /config GitVersion.yml
/opt/hostedtoolcache/GitVersion.Tool/5.8.1/x64/dotnet-gitversion /home/runner/work/drift-seasons/drift-seasons /output json /output buildserver /config GitVersion.yml
INFO [01/07/22 15:50:05:74] Working directory: /home/runner/work/drift-seasons/drift-seasons
INFO [01/07/22 15:50:05:75] Branch from build environment: refs/heads/develop
INFO [01/07/22 15:50:05:75] Project root is: /home/runner/work/drift-seasons/drift-seasons/
INFO [01/07/22 15:50:05:75] DotGit directory is: /home/runner/work/drift-seasons/drift-seasons/.git
INFO [01/07/22 15:50:05:75] Begin: Normalizing git directory for branch 'refs/heads/develop'
  INFO [01/07/22 15:50:05:79] One remote found (origin -> 'https://***.com/directivegames/drift-seasons').
  INFO [01/07/22 15:50:05:79] Skipping fetching, if GitVersion does not calculate your version as expected you might need to allow fetching or use dynamic repositories
  INFO [01/07/22 15:50:05:80] Updating local branch develop to point at 2bc6a97 {'jsonstring1': X}
  INFO [01/07/22 15:50:05:82] Creating local branch from remote tracking 'refs/remotes/origin/release/0.1'.
  INFO [01/07/22 15:50:05:84] HEAD points at branch 'refs/heads/develop'.
  INFO [01/07/22 15:50:05:84] End: Normalizing git directory for branch 'refs/heads/develop' (Took: 81.63ms)
  INFO [01/07/22 15:50:05:85] Begin: Loading version variables from disk cache
    INFO [01/07/22 15:50:05:85] Cache file /home/runner/work/drift-seasons/drift-seasons/.git/gitversion_cache/8D6572B3AC396D0F36C6D92CE1F25240751DE511.yml not found.
  INFO [01/07/22 15:50:05:85] End: Loading version variables from disk cache (Took: 0.94ms)
  INFO [01/07/22 15:50:05:94] Using latest commit on specified branch
  INFO [01/07/22 15:50:05:96] Running against branch: develop (2bc6a97 {'jsonstring1': X})
  INFO [01/07/22 15:50:05:96] Begin: Calculating base versions
    INFO [01/07/22 15:50:05:98] Fallback base version: 0.1.0 with commit count source 7ed6fe1aea2cd6351c25e769c11178760d94de29
    INFO [01/07/22 15:50:06:00] Found commit [2bc6a97 {'jsonstring1': X}] matching merge message format: Default
    INFO [01/07/22 15:50:06:03] Merge message 'Merge branch 'release/0.1' into develop': 0.1.0 with commit count source fc629b60cbccfa2066bb4e020757c5b172f2ace7
    INFO [01/07/22 15:50:06:03] Git tag '0.1.2': 0.1.2 with commit count source f965fed912096b435720ff9a28a3a66ecca991cd
    INFO [01/07/22 15:50:06:03] Git tag '0.1.1': 0.1.1 with commit count source d134b8891aebbb1f12327e1b0facf0b17fabe59f
    INFO [01/07/22 15:50:06:03] Git tag '0.1.0': 0.1.0 with commit count source 8ec3102a610a7eecddfaae9dc216422961c342ec
    INFO [01/07/22 15:50:06:04] Begin: Finding merge base between 'release/0.1' and 'develop'.
      INFO [01/07/22 15:50:06:05] Found merge base of f965fed Make losing players only lose half of the calculated amount as per request from game design
      INFO [01/07/22 15:50:06:05] Merge base was due to a forward merge, next merge base is d134b88 add player_id and season_id to stripped response when getting other players standing
      INFO [01/07/22 15:50:06:05] Merge base of release/0.1' and 'develop is d134b88 add player_id and season_id to stripped response when getting other players standing
    INFO [01/07/22 15:50:06:05] End: Finding merge base between 'release/0.1' and 'develop'. (Took: 9.51ms)
    INFO [01/07/22 15:50:06:05] Begin: Finding branch source of 'release/0.1'
    INFO [01/07/22 15:50:06:06] End: Finding branch source of 'release/0.1' (Took: 9.32ms)
    INFO [01/07/22 15:50:06:06] Release branch exists -> Version in branch name: 0.1.0 with commit count source d134b8891aebbb1f12327e1b0facf0b17fabe59f
    INFO [01/07/22 15:50:06:08] Found multiple base versions which will produce the same SemVer (0.2.0), taking oldest source for commit counting (Git tag '0.1.0')
    INFO [01/07/22 15:50:06:08] Base version used: Git tag '0.1.0': 0.1.0 with commit count source 8ec3102a610a7eecddfaae9dc216422961c342ec
  INFO [01/07/22 15:50:06:08] End: Calculating base versions (Took: 118.08ms)
  INFO [01/07/22 15:50:06:08] 15 commits found between 8ec3102 Merge branch 'develop' of ***.com:directivegames/drift-seasons into develop and 2bc6a97 {'jsonstring1': X}
  INFO [01/07/22 15:50:06:08] 15 commits found between 8ec3102 Merge branch 'develop' of ***.com:directivegames/drift-seasons into develop and 2bc6a97 {'jsonstring1': X}
  INFO [01/07/22 15:50:06:08] Begin: Getting version tags from branch 'refs/heads/develop'.
  INFO [01/07/22 15:50:06:09] End: Getting version tags from branch 'refs/heads/develop'. (Took: 2.01ms)
  INFO [01/07/22 15:50:06:09] Begin: Creating dictionary
  INFO [01/07/22 15:50:06:10] End: Creating dictionary (Took: 4.22ms)
  INFO [01/07/22 15:50:06:10] Begin: Storing version variables to cache file /home/runner/work/drift-seasons/drift-seasons/.git/gitversion_cache/8D6572B3AC396D0F36C6D92CE1F25240751DE511.yml
  INFO [01/07/22 15:50:06:13] End: Storing version variables to cache file /home/runner/work/drift-seasons/drift-seasons/.git/gitversion_cache/8D6572B3AC396D0F36C6D92CE1F25240751DE511.yml (Took: 33.97ms)
Executing GenerateSetVersionMessage for 'GitHubActions'.

Executing GenerateBuildLogOutput for 'GitHubActions'.
Writing version variables to $GITHUB_ENV file for 'GitHubActions'.
{
  "Major": 0,
  "Minor": 2,
  "Patch": 0,
  "PreReleaseTag": "alpha.15",
  "PreReleaseTagWithDash": "-alpha.15",
  "PreReleaseLabel": "alpha",
  "PreReleaseLabelWithDash": "-alpha",
  "PreReleaseNumber": 15,
  "WeightedPreReleaseNumber": 15,
  "BuildMetaData": null,
  "BuildMetaDataPadded": "",
  "FullBuildMetaData": "Branch.develop.Sha.2bc6a976ec8ad835cf6b15ad4b688233a172a244",
  "MajorMinorPatch": "0.2.0",
  "SemVer": "0.2.0-alpha.15",
  "LegacySemVer": "0.2.0-alpha15",
  "LegacySemVerPadded": "0.2.0-alpha0015",
  "AssemblySemVer": "0.2.0.0",
  "AssemblySemFileVer": "0.2.0.0",
  "FullSemVer": "0.2.0-alpha.15",
  "InformationalVersion": "0.2.0-alpha.15+Branch.develop.Sha.2bc6a976ec8ad835cf6b15ad4b688233a172a244",
  "BranchName": "develop",
  "EscapedBranchName": "develop",
  "Sha": "2bc6a976ec8ad835cf6b15ad4b688233a172a244",
  "ShortSha": "2bc6a97",
  "NuGetVersionV2": "0.2.0-alpha0015",
  "NuGetVersion": "0.2.0-alpha0015",
  "NuGetPreReleaseTagV2": "alpha0015",
  "NuGetPreReleaseTag": "alpha0015",
  "VersionSourceSha": "8ec3102a610a7eecddfaae9dc216422961c342ec",
  "CommitsSinceVersionSource": 15,
  "CommitsSinceVersionSourcePadded": "0015",
  "UncommittedChanges": 0,
  "CommitDate": "2022-01-07"
}
  INFO [01/07/22 15:50:06:25] Done writing 
Error: SyntaxError: Unexpected token ' in JSON at position 1
```

Rewriting the message restores sanity

**Comment by KristapsT** [comment]
Encountered the same issue with GitHub Actions. Some of commits in the history contain {} which is what I assume is the problem.

However, it looks like issues gets solved with using a newer version of `gittools/actions/gitversion/execute`

I can reproduce it with v0.9.9, but with v0.9.10 everything works fine.

**Comment by asbjornu** [comment]
As this is a bug in the task/action and not within GitVersion itself, I'm closing in favor of https://github.com/GitTools/actions/issues/730.
